collision.go

```diff
@@ -16,6 +16,16 @@ type SpaceComponent struct {
 	Height   float32
 }
 
+// Center positions the space component according to its center instead of its
+// top-left point (this avoids doing the same math each time in your systems)
+func (sc *SpaceComponent) Center(p Point) {
```

Comment:
Is it the name that you don't like? Maybe we should call it PositionCenter
The whole point of this method is to be able to move an entity by placing "it's center" to a desired point, but the name does not fully reflect that. An since this is just a helper I can place it anywhere in my own code but I felt like it could be interesting to reuse.